File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorder(self,root,ans):
        if root is not None:
            if root.left!=None:
                self.inorder(root.left,ans)
            
            ans.append(root.val)

            if root.right!=None:
                self.inorder(root.right,ans)
        return ans
            
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans=[]
        self.inorder(root,ans)
        return ans
            

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    void inorderTraversal(TreeNode* root, vector<int>& result) {

        if(root!=nullptr)
        {
            if(root->left!=nullptr)
                inorderTraversal(root->left,result);

            result.push_back(root->val);

            if(root->right!=nullptr)
                inorderTraversal(root->right,result);
        }

        return;
    }

public:
    vector<int> inorderTraversal(TreeNode* root)
    {
        vector<int> result;
        inorderTraversal(root,result);
        return result;
    }
};

